### ISSUE TITLE
2.0.1 uncaught exception

### ISSUE THREAD

**Issue by Ilovegym66** [iobroker-community-adapters/ioBroker.proxmox#297]
Since 2.0.1 this error appears:

```
2023-09-08 09:05:45.510 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.558 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.559 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (1): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.559 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.561 - error: proxmox.0 (1912059) Could not create nodes, please restart adapter: undefined
2023-09-08 09:05:45.596 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.596 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (2): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.597 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.599 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.599 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.599 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.599 - debug: proxmox.0 (1912059) clearing request timeout
2023-09-08 09:05:45.600 - info: proxmox.0 (1912059) terminating
2023-09-08 09:05:45.600 - warn: proxmox.0 (1912059) Terminated (UNCAUGHT_EXCEPTION): Without reason
2023-09-08 09:05:45.633 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.633 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (3): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.634 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.635 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.635 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.635 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.636 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.636 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.636 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.669 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.669 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (4): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.669 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.670 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.671 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.671 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.705 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.705 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (5): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.706 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.707 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.707 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.707 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.745 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.746 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (6): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.746 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.747 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.747 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.748 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.781 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.781 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (7): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.781 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.782 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.783 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.783 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.816 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.816 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (8): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.816 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.818 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.818 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.818 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.850 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.851 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (9): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.851 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.852 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.853 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.853 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.886 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.886 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (10): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.886 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.888 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.888 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.888 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.923 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.924 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (11): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.924 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.925 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.926 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.926 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:45.963 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:45.963 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (12): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:45.963 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:45.965 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:45.965 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:45.965 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:46.003 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:46.003 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (13): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:46.003 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:46.005 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:46.005 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:46.005 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:46.041 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:46.041 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (14): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:46.041 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:46.043 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:46.043 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:46.043 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:46.083 - error: proxmox.0 (1912059) received 500 response from /nodes/hpg380/disks/smart?disk=/dev/sda with content: {"data":null}
2023-09-08 09:05:46.083 - info: proxmox.0 (1912059) Use Next Proxmox Host because of communication failure (15): https://192.168.0.210:8006/api2/json
2023-09-08 09:05:46.083 - debug: proxmox.0 (1912059) starting request to /nodes/hpg380/disks/smart?disk=/dev/sda ...
2023-09-08 09:05:46.085 - error: proxmox.0 (1912059) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-09-08 09:05:46.085 - error: proxmox.0 (1912059) unhandled promise rejection: undefined
2023-09-08 09:05:46.085 - error: proxmox.0 (1912059) undefined
2023-09-08 09:05:46.101 - info: proxmox.0 (1912059) terminating
2023-09-08 09:05:46.130 - error: host.iobrokerS2 instance system.adapter.proxmox.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
2023-09-08 09:05:46.131 - info: host.iobrokerS2 Restart adapter system.adapter.proxmox.0 because enabled
```

**Comment by Ilovegym66** [comment]
Additional:
all other data are in the objects, that's fine. Crashed only on the HPDL380G8 Server, on the Nuc the Adapter works fine.
here my HDD Config of the HP Server:

ilovegym@hpg380:~$ sudo fdisk -l
Disk /dev/sda: 8.19 TiB, 9001677840384 bytes, 17581402032 sectors
Disk model: LOGICAL VOLUME
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes
Disklabel type: gpt
Disk identifier: 7007876B-B312-454A-BCFA-C0F47BE58D75

Device       Start         End     Sectors  Size Type
/dev/sda1       34        2047        2014 1007K BIOS boot
/dev/sda2     2048     2099199     2097152    1G EFI System
/dev/sda3  2099200 17581401998 17579302799  8.2T Linux LVM

Partition 1 does not start on physical sector boundary.


Disk /dev/mapper/pve-swap: 4 GiB, 4294967296 bytes, 8388608 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/mapper/pve-root: 128 GiB, 137438953472 bytes, 268435456 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/sdb: 256 MiB, 268435456 bytes, 524288 sectors
Disk model: LUN 00 Media 0
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x00000046

Device     Boot Start    End Sectors  Size Id Type
/dev/sdb1          63 514079  514017  251M  c W95 FAT32 (LBA)


Disk /dev/mapper/pve-vm--100--disk--0: 20 GiB, 21474836480 bytes, 41943040 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/mapper/pve-vm--101--disk--0: 32 GiB, 34359738368 bytes, 67108864 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/mapper/pve-vm--100--disk--1: 250 GiB, 268435456000 bytes, 524288000 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/mapper/pve-vm--104--disk--0: 16 GiB, 17179869184 bytes, 33554432 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/mapper/pve-vm--105--disk--0: 8 GiB, 8589934592 bytes, 16777216 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/mapper/pve-vm--103--disk--1: 32 GiB, 34359738368 bytes, 67108864 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes


Disk /dev/mapper/pve-vm--102--disk--0: 32 GiB, 34359738368 bytes, 67108864 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 262144 bytes / 786432 bytes

**Comment by klein0r** [comment]
Can you share the disk overview of the proxmox ui? I don't know why `/nodes/hpg380/disks/smart?disk=/dev/sda` leads to HTTP 500. Since the device exists:

> Disk /dev/sda: 8.19 TiB, 9001677840384 bytes, 17581402032 sectors

**Comment by klein0r** [comment]
Please test with 2.0.2

**Comment by Ilovegym66** [comment]
fantastic! fixed with 2.0.2, thank you very much for the fast support!
